java/code/src/com/redhat/rhn/taskomatic/task/ReportDBHelper.java

```diff
@@ -196,4 +199,70 @@ public void analyzeReportDb(Session session) {
         var m = ModeFactory.getCallableMode(session, "GeneralReport_queries", "analyze_reportdb");
         m.execute(new HashMap<>(), new HashMap<>());
     }
+
+    /**
+     * Check if a specific user is configured in the database
+     * @param session the session
+     * @param username the username to search for
+     * @return return true when the user exists, otherwise return false
+     */
+    public boolean hasDBUser(Session session, String username) {
+        final String sqlStatement = "SELECT usename FROM pg_catalog.pg_user";
+        var m = new GeneratedSelectMode("select.pg_catalog.user", session, sqlStatement, Collections.emptyList());
+        DataResult<Map<String, String>> result = m.execute();
+        return result.stream().map(e -> e.get("usename")).anyMatch(n -> n.equalsIgnoreCase(username));
+    }
+
+    /**
+     * Create a new user in the given database and grant permissions
+     * @param session the session
+     * @param dbName the db name
+     * @param username the new username
+     * @param password the new password
+     */
+    public void createDBUser(Session session, String dbName, String username, String password) {
+        final String sql = MessageFormat.format("""
+            CREATE ROLE {0} WITH LOGIN PASSWORD ''{1}'' NOSUPERUSER INHERIT NOCREATEDB NOCREATEROLE NOREPLICATION;
+            GRANT CONNECT ON DATABASE {2} TO {0};
+            GRANT USAGE ON SCHEMA public TO {0};
+            GRANT SELECT ON ALL TABLES IN SCHEMA public TO {0};
+            GRANT SELECT ON ALL SEQUENCES IN SCHEMA public TO {0};
+            """, username, password, dbName);
```

Comment:
JFYI, we don't strictly need to use `MessageFormat.format()` in these cases, we can simply use java 17 `"".formatted()` with explicit argument index  `%1$s`, `%2$s`

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Formatter.html#syntax

This creates the same string:

```java
        final String sql = """
            CREATE ROLE %1$s WITH LOGIN PASSWORD '%2$s' NOSUPERUSER INHERIT NOCREATEDB NOCREATEROLE NOREPLICATION;
            GRANT CONNECT ON DATABASE %3$s TO %1$s;
            GRANT USAGE ON SCHEMA public TO %1$s;
            GRANT SELECT ON ALL TABLES IN SCHEMA public TO %1$s;
            GRANT SELECT ON ALL SEQUENCES IN SCHEMA public TO %1$s;
            """.formatted(username, password, dbName);
```